src/Aspire.Hosting/Dcp/ResourceLogSource.cs

```diff
@@ -35,13 +37,19 @@ public async IAsyncEnumerator<LogEntryList> GetAsyncEnumerator(CancellationToken
             SingleWriter = false
         });
 
+        var startupStdoutStreamTask = Task.Run(() => StreamLogsAsync(startupStdoutStream, isError: false), cancellationToken);
+        var startupStderrStreamTask = Task.Run(() => StreamLogsAsync(startupStderrStream, isError: true), cancellationToken);
         var stdoutStreamTask = Task.Run(() => StreamLogsAsync(stdoutStream, isError: false), cancellationToken);
         var stderrStreamTask = Task.Run(() => StreamLogsAsync(stderrStream, isError: true), cancellationToken);
 
         // End the enumeration when both streams have been read to completion.
 
         async Task WaitForStreamsToCompleteAsync()
         {
+            // We want all the startup streams to be completed before we render anything
```

Comment:
This method just waits for the tasks to complete, it does not affect how output is consumed in any way, so it is completely fine to add the startup tasks to the list and do a single `WaitAll()` call.

Reply:
So it is possible to get interleaved output. I probably need to not start the streaming of the stdout and stderr streams until the startup streams have completed.

In practice it seems to only happen if you are setting breakpoints in the code here to debug things. But the fact that interleaving can occur makes me think we should fix it :)

Reply:
I've reverted it back to where it was. I've decided that whilst this is a problem in theory, its not a problem in practice. My main goal in this PR is to get the build/pull logs showing.

It would be great if you and @danegsta can get at getting the build logs streaming so folks can get some kind of visual queue as to what is happening (so it doesn't look like it has hung).